Python/pylifecycle.c

```diff
@@ -2127,15 +2127,15 @@ _Py_Finalize(_PyRuntimeState *runtime)
     _PyGC_CollectIfEnabled();
 #endif
 
-    /* Disable tracemalloc after all Python objects have been destroyed,
-       so it is possible to use tracemalloc in objects destructor. */
-    _PyTraceMalloc_Fini();
-
     /* Finalize any remaining import state */
     // XXX Move these up to where finalize_modules() is currently.
     _PyImport_FiniCore(tstate->interp);
     _PyImport_Fini();
 
+    /* Disable tracemalloc after all Python objects have been destroyed,
+       so it is possible to use tracemalloc in objects destructor. */
+    _PyTraceMalloc_Fini();
```

Comment:
I don't think that this fix is enough. Some objects can be deleted later, in finalize_interp_clear(). So I wrote a different fix to restore the old behavior: PR gh-129191.